### PR TITLE
Porting EfficientNets to KerasCV

### LINKED ISSUE #551: EfficientNet architecture

### PR DESCRIPTION
**Pull request by AdityaKane2001**
@sayakpaul @LukeWood @ianjjohnson 

Ported over EfficientNets (V1 and V2) from keras. Made changes wherever necessary to stick to the KerasCV code style.

/auto Closes #551

### CODE REVIEW

**Comment by AdityaKane2001** [comment]
@LukeWood 

The test process was killed: 
image

**Comment by innat** [comment]
There is already an active PR for eff-v1. 
https://github.com/keras-team/keras-cv/pull/570

**Comment by LukeWood** [comment]
@AdityaKane2001 try rebasing this PR!

File: keras_cv/models/efficientnet_v1.py [hunk 1]
```diff
@@ -0,0 +1,733 @@
...
+            - `None` means that the output of the model will be
+                the 4D tensor output of the
+                last convolutional layer.
+            - `avg` means that global average pooling
+                will be applied to the output of the
+                last convolutional layer, and thus
+                the output of the model will be a 2D tensor.
+            - `max` means that global max pooling will
+                be applied.
+        num_classes: optional number of num_classes to classify images
```
Comment (LukeWood) [inline keras_cv/models/efficientnet_v1.py:366]: we renamed `num_classes` to `classes` recently for API consistency, lets make that change here!

File: keras_cv/models/efficientnet_v1.py [hunk 2]
```diff
@@ -0,0 +1,733 @@
...
+    Raises:
+      ValueError: in case of invalid argument for `weights`,
+        or invalid input shape.
+      ValueError: if `classifier_activation` is not `softmax` or `None` when
+        using a pretrained top layer.
+    """
+    if blocks_args == "default":
+        blocks_args = DEFAULT_BLOCKS_ARGS
+
+    if weights and not tf.io.gfile.exists(weights):
```
Comment (LukeWood) [inline keras_cv/models/efficientnet_v1.py:385]: please use:

https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/weights.py#L17

now!

**Review by LukeWood — CHANGES_REQUESTED** [review]
Overall, looks great!  Just a minor nit or two... :)

**Pull request closed without merging**